docker_mounts.go

```diff
@@ -85,6 +100,40 @@ func (s DockerTmpfsMountSource) GetTmpfsOptions() *mount.TmpfsOptions {
 	return s.TmpfsOptions
 }
 
+// DockerImageMountSource is a mount source for an image
+type DockerImageMountSource struct {
```

Comment:
what about just `ImageMountSource`?

Reply:
I used this for keeping consistency with the existing types. As mentioned in the follow-ups section:

> I do not like the code for mounting volumes and tmpfs, as it's hard to follow, and in my opinion there are too many abstractions. So I'd advocate for starting a discussion on how to do a hard refactor for that part.

So I'm open to start a rewrite of that layer

Reply:
I'd go with the simple naming too.

Reply:
Agreed there must be a simpler abstraction for this seems there's multiple layers just adding complexity.